cmflib/commands/metadata/pull.py

```diff
@@ -0,0 +1,129 @@
+###
+# Copyright (2022) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+#!/usr/bin/env python3
+import argparse
+import json
+import os
+from cmflib import merger
+from cmflib import cmfquery
+from cmflib.cli.command import CmdBase
+from cmflib.cli.utils import read_cmf_config, find_root
+from cmflib.request_mlmdserver import server_interface
+
+# This class pulls mlmd file from cmf-server
+class CmdMetadataPull(CmdBase):
+    def run(self):
+        config_file = ".cmfconfig"
+        url = "http://127.0.0.1:80"
+        if os.path.exists(find_root(config_file)):
+            url = read_cmf_config(os.path.join(find_root(config_file), config_file)).split("=")[1]
+        directory_to_dump = ""
+        data = ""
+        cmd = "pull"
+        mlmd_data = ""
+        status = 0
+        execution_flag = 0
+        if self.args.file_path:  # setting directory where mlmd file will be dumped
+            directory_to_dump = self.args.file_path
+        else:
+            directory_to_dump = os.getcwd()
+        output = server_interface.call_mlmd_pull(
+            url, self.args.pipeline_name
```

Comment:
Here if execution id is given , we should pull only for that execution id, not the entire data. We need to be efficient in data transfer. So we cannot pull the entire data